server/market/bookrouter.go

```diff
@@ -344,6 +344,7 @@ out:
 					epochNote.BookOrderNote = *marketOrderToMsgOrder(o, book.name)
 					epochNote.OrderType = msgjson.MarketOrderNum
 				case *order.CancelOrder:
+					epochNote.BookOrderNote = *cancelOrderToMsgOrder(o, book.name)
```

Comment:
This is the important fix.  Some fields of the msgjson.BookOrderNote were left zero, including order id.